### PR TITLE
changed directory of update user APIs from `auth` to `user`

### LINKED ISSUE #40: Update Info APIs should be moved to a different directory to avoid confusion
here's an idea:
what if all the APIs under the `auth` folder which requires `access_token` are shifted to `user` folder... like this:
I think it will remove a lot of confusion.

```
    - auth
        - index.js
        - signup.js
        - login.js
        - resetPassword.js
        - forgotPassword.js
        - verifyEmail.js
        - verifyNewEmail.js
        - isCorrect.js
    -user
        - index.js
        - getUser.js
        - updateUser.js
        - updatePassword.js
        - updateProfileImage.js
``` 

Also, these upadteInfo APIs really dont come under `auth` section

_Originally posted by @ridhishjain in https://github.com/Cyber-Labs/cl-judge/pull/39#issuecomment-711126014_

### PR DESCRIPTION
**Pull request by ridhishjain**
resolves #40

### CODE REVIEW

File: server/models/auth/index.js [hunk 1]
```diff
@@ -3,21 +3,13 @@ const verifyEmail = require('./verifyEmail')
...
   signup,
   verifyEmail,
   login,
   forgotPassword,
   resetPassword,
-  updatePassword,
-  updateUser,
   verifyNewEmail,
-  getUser,
-  uploadProfileImage,
```
Comment (ridhishjain) [inline server/models/auth/index.js:22]: all these moved to `models/user`.

File: server/models/auth/login.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable no-async-promise-executor */
 const fs = require('fs')
 const jwt = require('jsonwebtoken')
-const isCorrect = require('./isCorrect')
+const { isPasswordCorrect } = require('../utils')
```
Comment (ridhishjain) [inline server/models/auth/login.js:4]: `isCorrect.js` renamed to `isPasswordCorrect` and moved to `models/utils`

File: server/models/user/updateProfileImage.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ const { pool } = require('../database')
  *
  */
 
-function uploadProfileImage({ file, username }) {
+function updateProfileImage({ file, username }) {
```
Comment (ridhishjain) [inline server/models/user/updateProfileImage.js:12]: `uploadProfileImage` changed to `updateProfileImage`

File: server/routes/auth/index.js [hunk 4]
```diff
@@ -3,21 +3,13 @@ const verifyEmailRouter = require('./verifyEmail')
...
   signupRouter,
   verifyEmailRouter,
   loginRouter,
   forgotPasswordRouter,
   resetPassowrdRouter,
-  updatePasswordRouter,
-  updateUserRouter,
   verifyNewEmailRouter,
-  getUserRouter,
-  uploadProfileImage,
```
Comment (ridhishjain) [inline server/routes/auth/index.js:22]: all these moved to `routes/user`

File: server/routes/index.js [hunk 5]
```diff
@@ -11,17 +11,19 @@ app.use(
...
-router.use('/auth', authRouter.updatePasswordRouter)
-router.use('/auth', authRouter.updateUserRouter)
-router.use('/auth', authRouter.uploadProfileImage)
 router.use('/auth', authRouter.verifyNewEmailRouter)
-router.use('/user', authRouter.getUserRouter)
+
+router.use('/user', userRouter.getUserRouter)
+router.use('/user', userRouter.updateUserRouter)
+router.use('/user', userRouter.updatePasswordRouter)
+router.use('/user', userRouter.updateProfileImageRouter)
```
Comment (ridhishjain) [inline server/routes/index.js:26]: route changed

File: server/schema/auth/index.js [hunk 6]
```diff
@@ -18,7 +16,5 @@ module.exports = {
   loginSchema,
   forgotPassowrdSchema,
   resetPasswordSchema,
-  updatePasswordSchema,
-  updateUserSchema,
```
Comment (ridhishjain) [inline server/schema/auth/index.js:22]: all these moved to `schema/user`

**Review by cjchirag7 — APPROVED** [review]

**Comment by cjchirag7** [comment]
We can also move verifyNewMail to user

**Comment by cjchirag7** [comment]
Anywas, it's fine, with these changes only.

**Pull request merged**